R/covariance_plot.R

```diff
@@ -54,3 +56,44 @@ h_get_timepoint_vars <- function(vcov_matrix,
     col_time = col_time
   )
 }
+
+#' Vectorization helper for Covariance Plot
+#'
+#' @description This function vectorizes the upper-diagonal elements of a symmetric matrix (e.g. the covariance matrix)
+#' and obtains the lag and time distance between pairs of observations if the time values are part of
+#' the names or are part of the matrix column/row names.
+#'
+#' @param vcov_matrix (`matrix`)\cr name of the input symmetric matrix.
+#' @return A data frame with the upper-diagonal elements or a covariance or correlation
+#' matrix. In the context of repeated measures, this matrix contains the association between pairs
+#' of measurements taken at different time points. It contains the following columns: a column
+#' with the upper-diagonal elements of vcov_matrix, the difference between column time
+#' and row time, the lag and the row and column ranks.
+#'
+#' @examples
+#' vcov_matrix <- matrix(
+#'   c(49, 24, 12, 23, 24, 35, 11, 20, 12, 11, 24, 14, 23, 20, 14, 107),
+#'   nrow = 4, ncol = 4,
+#'   dimnames = list(
+#'     c("VIS1", "VIS2", "VIS3", "VIS4"),
+#'     c("VIS1", "VIS2", "VIS3", "VIS4")
+#'   )
+#' )
+#' h_vectorization(vcov_matrix, string = "VIS")
```

Comment:
```suggestion
```
Best is to just omit the example. Basically, internal functions cannot have examples.